### ISSUE TITLE
Date picker may break with long month names at 320px width

### ISSUE THREAD

**Issue by thisisdano** [uswds/uswds#4761]
We need to support 320 CSS pixel device width to support WCAG 2.1. The date picker works at 320px wide, but it may fail when on a 320px-wide page with standard left and right padding (2rem). This is exacerbated by putting the field group in an error state, which adds another rem or so of left padding.

In real-life usage, this means that the date picker needs to support a width of about 224-256px. This width breaks the date picker display with long months like `November` and `September`.

<img width="321" alt="Date picker display breaks at very narrow widths" src="https://user-images.githubusercontent.com/11464021/171675860-2d3f0672-269f-428a-8ac7-1f54d1b94ac9.png">

## Steps to reproduce
- Visit the Date picker page at https://designsystem.digital.gov/components/date-picker/
- In the browser inspector, resize to 320px wide
- Open the Date picker and navigate to `September`
